api/endpoints/members.py

```diff
@@ -447,36 +448,73 @@ class AwsAccessEdcCredentials(Resource):
     """
     @login_required
     def get(self, endpoint_uri):
-
-        s = requests.Session()
         maap_user = get_authorized_user()
 
-        if maap_user is None:
+        if not maap_user:
             return Response('Unauthorized', status=401)
-        else:
-            urs_token = db.session.query(Member_db).filter_by(id=maap_user.id).first().urs_token
-            s.headers.update({'Authorization': f'Bearer {urs_token},Basic {settings.MAAP_EDL_CREDS}',
-                              'Connection': 'close'})
 
-            endpoint = parse.unquote(endpoint_uri)
-            login_resp = s.get(
-                endpoint, allow_redirects=False
-            )
-            login_resp.raise_for_status()
+        creds = get_edc_credentials(endpoint_uri, maap_user)
+
+        response = jsonify(
+            accessKeyId=creds['accessKeyId'],
+            secretAccessKey=creds['secretAccessKey'],
+            sessionToken=creds['sessionToken'],
+            expiration=creds['expiration']
+        )
+
+        response.headers.add('Access-Control-Allow-Origin', '*')
+
+        return response
+
+
+def creds_expiration_utc_timestamp(_key, creds, _now) -> float:
+    """Return the expiration time of a credentials object converted to the
+    number of seconds (fractional) since the epoch in UTC, minus 5 minutes."""
+
+    expiration = creds.get('expiration')
+
+    try:
+        expiration_dt = datetime.strptime(expiration, "%Y-%m-%d %H:%M:%S%z")
+    except (TypeError, ValueError):
+        expiration_dt = datetime.now()
 
-            edl_response = s.get(url=login_resp.headers['location'])
-            json_response = json.loads(edl_response.content)
+    expiration_dt_utc = expiration_dt.astimezone(timezone.utc)
 
-            response = jsonify(
-                accessKeyId=json_response['accessKeyId'],
-                secretAccessKey=json_response['secretAccessKey'],
-                sessionToken=json_response['sessionToken'],
-                expiration=json_response['expiration']
-            )
+    # Subtract 5 minutes for "wiggle room"
+    return (expiration_dt_utc - timedelta(minutes=5)).timestamp()
```

Comment:
Given the timeout duration is 60 minutes, I might increase the wiggle room up to 30 minutes to account for long running operations.

Reply:
I've updated the logic to halve the expiry time, however long it is, so that we don't make any assumptions about the duration.